src/linear_algebra/bilinear_form.lean

```diff
@@ -1694,6 +1712,20 @@ begin
       linear_equiv.symm_apply_apply]
 end
 
+lemma nondegenerate_iff_det_ne_zero (B : bilin_form A M₃)
+  (b : basis ι A M₃) : B.nondegenerate ↔ ((bilin_form.to_matrix b) B).det ≠ 0 :=
```

Comment:
It might be nicer to prove this as:
```suggestion
lemma to_bilin'_nondegenerate_iff_to_bilin_nondegenerate {M : matrix ι ι A}
  (b : basis ι A M₃) : (matrix.to_bilin' M).nondegenerate ↔ (matrix.to_bilin b M).nondegenerate :=
begin
  sorry  -- is this easier or harder than the proof you have right now?
end

lemma nondegenerate_iff_det_ne_zero (B : bilin_form A M₃)
  (b : basis ι A M₃) : B.nondegenerate ↔ ((bilin_form.to_matrix b) B).det ≠ 0 :=
begin
  revert B,
  rw (bilin_form.to_matrix b).symm.surjective.forall,
  intro M,
  rw [to_matrix_symm, to_matrix_to_bilin, ←to_bilin.nondegenerate_iff_det_ne_zero',
    ←to_bilin'_nondegenerate_iff_to_bilin_nondegenerate],
end
```
It feels like we're missing the API linking `to_matrix` with `to_matrix'` and `to_bilin` with `to_bilin'`. If we had that we shouldn't have to prove the same thing so often.

Reply:
Here's one version of that proof:
```
lemma congr_nondegenerate_iff
  {R} [comm_semiring R] [module R M] [module R M₂] {B : bilin_form R M₂} (e : M₂ ≃ₗ[R] M):
  (congr e B).nondegenerate ↔ B.nondegenerate :=
e.surjective.forall.trans $ forall_congr $ λ m, imp_congr
  (e.surjective.forall.trans $ forall_congr $ λ m, by
    rw [congr_apply, e.symm_apply_apply, e.symm_apply_apply])
  (linear_equiv.map_eq_zero_iff _)

lemma to_bilin'_nondegenerate_iff_to_bilin_nondegenerate {M : matrix ι ι A}
  (b : basis ι A M₃) : (matrix.to_bilin' M).nondegenerate ↔ (matrix.to_bilin b M).nondegenerate :=
(congr_nondegenerate_iff b.equiv_fun.symm).symm
```

Reply:
PR'd some dependencies as #10362, there were a bunch of API holes.